spec/integration/wrap_spec.rb

```diff
@@ -86,6 +86,39 @@
           let(:name) { :assignee }
         end
       end
+
+      context 'using association with a view' do
+        before do
+          conf.relation(:users) do
+            auto_map false
+
+            schema(infer: true) do
+            end
+
+            def with_extra_attributes
+              select { `'testing'`.as(:extra_attribute) }
+            end
+          end
+
+          conf.relation(:tasks) do
+            auto_map true
+
+            schema(infer: true) do
+              associations do
+                belongs_to :users, view: :with_extra_attributes
+              end
+            end
+          end
+        end
+
+        it 'includes the extra attributes' do
+          result = tasks.wrap(:users).to_a
+
+          result.each do |task|
+            expect(task[:users]).to eql(extra_attribute: 'testing')
+          end
```

Comment:
It would be better to materialize this relation and compare it with the expected resulting structure. If, for whatever reason, `result` is empty, that `each` won't do a thing and the spec will continue to pass.